### ISSUE TITLE
"Restore Recent Closed File" text disappears from menu

### ISSUE THREAD

**Issue by jonandr** [notepad-plus-plus/notepad-plus-plus#346]
Open N++
Empty recent file list
Close N++
Open N++ again
Open a file
Close the file
=>
"Restore recent closed file" is no longer visible in the menu.

Have not been able to figure out why, maybe someone else knows.

![x](https://cloud.githubusercontent.com/assets/1019727/8385708/9148f6f0-1c4a-11e5-8bf5-8d390e51b761.png)

**Comment by Yaron10** [comment]
Hello Andreas,

It happens in various different cases.
https://sourceforge.net/p/notepad-plus/discussion/331753/thread/6557b445/#3de6
https://sourceforge.net/p/notepad-plus/discussion/331753/thread/6177edd4/#aab7 

***
May I request your help?

Using the Incremental Search Bar's text field, I often find myself "Select All" and then "Paste".
I'd like to make that field select all on click.

I downloaded NP++ source code a couple of days ago, and my experience is little to none.
- I assume I need to add a listener to the control "EDITTEXT  IDC_INCFINDTEXT" getting the focus, and then send it a "Select All" message .
I've been trying to implement this in "FindReplaceDlg.cpp" but, so far, without success.

I suppose that for you it might be a five-minute work.
If indeed this is the case, I'd be very grateful for your help.

Best regards.

**Comment by jonandr** [comment]
@Yaron10 After line 2871 in FindReplaceDlg.cpp, add the following:

```c++
else if (HIWORD(wParam) == EN_SETFOCUS)
{
	::PostMessage(::GetDlgItem(_hSelf, IDC_INCFINDTEXT), EM_SETSEL, 0, -1);
	break;
}
```

**Comment by Yaron10** [comment]
Hello Andreas,

Thank you so much! Excellent.
I really appreciate it.

Best regards.

**Comment by Yaron10** [comment]
Hello Andreas,

A minor issue.
Ignore if you don't find it worth your while.

Open the search bar and type some characters.
Click somewhere in the editor.
*Right-click* in search bar's text field.

Result: the text is selected but the commands "Copy", "Paste" etc. are grayed-out.

If you now repeat steps 2 and 3, those commands are available.

Thanks again.

**Comment by Yaron10** [comment]
Thanks for fixing this bug.
Can the string still be localized?

**Comment by Yaron10** [comment]
Clear the Recent Files list.
Close NP++.
In Windows Explorer (Win 7), open a file associated with NP++.
Close the file.

Result: the text is missing.

**Comment by donho** [comment]
@Yaron10 Cannot reproduce it under my Windows 7. Works all fine to me.

**Comment by Yaron10** [comment]
I've rebuilt the solution, and it's working properly now
My sincere apologies and thanks again.

**Comment by Rikk** [comment]
This "fix" is causing issues with translations: https://github.com/notepad-plus-plus/notepad-plus-plus/pull/542#issuecomment-127525831

Imo, the problem for it disappearing is some failure in re-adding the menu items after then have been removed, maybe at this part of `updateMenu()` function: https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/PowerEditor/src/lastRecentFileList.cpp#L163

**Comment by Rikk** [comment]
This "fix" is causing issues with translations: https://github.com/notepad-plus-plus/notepad-plus-plus/pull/542#issuecomment-127525831

Imo, the problem for it disappearing is some failure in re-adding the menu items after then have been removed, maybe at this part of `updateMenu()` function: https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/PowerEditor/src/lastRecentFileList.cpp#L160-L164

Therefore, imo, just setting a default value for it (`TEXT("Restore Recent Closed File")`) instead of `NULL`, does not fix the root cause of the problem (previous paragraph).

**Comment by Yaron10** [comment]
@Rikk,

Could you please create a patch?

Thanks.

**Comment by Rikk** [comment]
Sorry, I have no clue how to fix. I'm still slowly learning about Npp very complex source-code.

Further I will go for now is a supposition that @donho forgot something when introduced "Restore last closed file" feature.

**Comment by Yaron10** [comment]
@Rikk,

Thanks for replying. I appreciate it.
And good luck with learning the source-code. :)

**Comment by Rikk** [comment]
Thanks!

But my goal are easy to fix things that itch me, like #633 #637.

Only thing I could fix in this case are a couple duplicated lines https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/PowerEditor/src/lastRecentFileList.cpp#L73 and https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/PowerEditor/src/lastRecentFileList.cpp#L85

**Comment by Yaron10** [comment]
Your fixes for #633 #637 are really good. Especially the Gripper.
Keep up the good work.

Many thanks.
